app/models/robot_world.rb

```diff
@@ -1,56 +1,80 @@
+                                        date_hired, department, avatar)
+                     VALUES (\"#{robo[:name]}\", \"#{robo[:city]}\", \"#{robo[:state]}\",
+                            \"#{robo[:birthdate]}\", \"#{robo[:date_hired]}\",
+                            \"#{robo[:department]}\", \"#{robo[:avatar]}\");"
+                    )
   end
 
-  Robot = Struct.new(:id, :name, :city, :state,
-                     :birthdate,
-                     :date_hired, :department, :avatar)
+  def new_robot(robot_details)
+    Robot.new(robot_details)
+  end
 
-  def new_robot(id, name, city, state, birthday, date_hired, department, avatar)
-    Robot.new(id, name, city, state, birthday, date_hired, department, avatar)
+  def format_date(date)
+    unless date.nil?
+      "#{date["month"].to_s.rjust(2,"0")}/#{date["day"].to_s.rjust(2,"0")}/#{date["year"].to_s.rjust(4,"0")}"
+    end
   end
 
   def all
-    world.transaction do
-      world['robots'] ||= []
-    end
+    database.execute("SELECT * FROM robots;")
   end
 
   def find(robot_id)
-    all.find{|robot| robot.id == robot_id}
-  end
-
-  def update(robot_id, robot_data)
-    world.transaction do
-      targeted_robot = world['robots'].find{|robot| robot.id == robot_id}
-      targeted_robot[:name] = robot_data[:name]
-      targeted_robot[:city] = robot_data[:city]
-      targeted_robot[:state] = robot_data[:state]
-      targeted_robot[:birthdate] = robot_data[:birthdate]
-      targeted_robot[:date_hired] = robot_data[:date_hired]
-      targeted_robot[:department] = robot_data[:department]
-      targeted_robot[:avatar] = robot_data[:avatar]
-    end
+    robot = new_robot(all.find do |robot| 
+      robot["id"] == robot_id
+    end.delete_if{|k,v| k.is_a?(Fixnum)})
+  end
+
+  def update(robot_id, new_robot_data)
+    database.execute("UPDATE robots 
+                     SET name=? 
+                     WHERE id=?;", [new_robot_data["name"], robot_id]) unless new_robot_data["name"].nil?
+    database.execute("UPDATE robots 
+                     SET city=? 
+                     WHERE id=?;", [new_robot_data["city"], robot_id]) unless new_robot_data["city"].nil?
+    database.execute("UPDATE robots 
+                     SET state=? 
+                     WHERE id=?;", [new_robot_data["state"], robot_id]) unless new_robot_data["state"].nil?
+    database.execute("UPDATE robots 
+                     SET birthdate=? 
+                     WHERE id=?;", [format_date(new_robot_data["birthdate"]), robot_id]) unless new_robot_data["birthdate"].nil?
+    database.execute("UPDATE robots 
+                     SET date_hired=? 
+                     WHERE id=?;", [format_date(new_robot_data["date_hired"]), robot_id]) unless new_robot_data["date_hired"].nil?
+    database.execute("UPDATE robots 
+                     SET department=? 
+                     WHERE id=?;", [new_robot_data["department"], robot_id]) unless new_robot_data["department"].nil?
+    database.execute("UPDATE robots 
```

Comment:
each of these `database.execute("UPDATE robots")` statements are pretty much identical. how can you create a method that makes this behavior more DRY?